patches/chromium/printing.patch

```diff
@@ -315,24 +315,27 @@ index c976fb2a814e4ff09650982034371e40d1ab77bc..c9115c7bbead588f3099bb194eb293b0
  
    std::unique_ptr<PrintSettings> print_settings =
        PrintSettingsFromJobSettings(job_settings);
-@@ -704,6 +751,16 @@ void PrintViewManagerBase::UpdatePrintSettings(
+@@ -704,7 +751,18 @@ void PrintViewManagerBase::UpdatePrintSettings(
      }
    }
  
+-#if BUILDFLAG(IS_WIN)
 +  std::unique_ptr<PrinterQuery> query =
 +      queue_->CreatePrinterQuery(GetCurrentTargetFrame()->GetGlobalId());
 +  auto* query_ptr = query.get();
++  auto job_settings_copy = job_settings.Clone();
 +  query_ptr->SetSettings(
-+      job_settings.Clone(),
++      std::move(job_settings_copy),
```

Comment:
Semantically passing `job_settings.Clone()` directly does not really make a difference from putting it in a local value and then moving it. It seems that a memory corruption happened somewhere and this code changed memory layout and hid the bug. Do you have a full stack trace of the crash?

Reply:
@zcbenz i didn't think it did either - it's consistent though and the only stack trace seen is the bad optional access crash, which happens [here](https://source.chromium.org/chromium/chromium/src/+/refs/heads/main:chrome/browser/printing/printer_query.cc;l=330-331;drc=04e8ef07905d9f4b29879f49ed8b338b25c2b092?q=SetSettings%20printer_query&sq=package:chromium&type=cs). `has_value` is always false when finding  `kSettingPrinterType` when calling `job_settings.Clone()`